diff --git a/node_modules/github-version-checker/lib/check.js b/node_modules/github-version-checker/lib/check.js
index 7f00060..cf0e231 100644
--- a/node_modules/github-version-checker/lib/check.js
+++ b/node_modules/github-version-checker/lib/check.js
@@ -88,7 +88,6 @@ const rest = (options, callback) => {
 
       if (found) {
         const mapped = schemeMapper[options.fetchTags ? 'tag' : 'release'](version);
-        console.log(mapped);
         callback(null, mapped);
       } else {
         callback(null, null);
